kernel/src/main/java/org/kframework/parser/inner/ParseInModule.java

```diff
@@ -445,7 +445,7 @@ private Tuple2<Either<Set<KEMException>, Term>, Set<KEMException>> parseStringTe
           boolean equalRight =
               rez.isRight() && z3Rez.isRight() && rez.right().get().equals(z3Rez.right().get());
           if (!(bothLeft || equalRight)) {
-            throw KEMException.criticalError("Z3 and SimpleSub algorithms differ!");
+            throw typeInferenceCheckError(rez3, z3Rez, rez);
```

Comment:
Do you have any examples of such differences?
It would be interesting to document them and add regression tests.

Reply:
I'm not aware of any actual differences - the improved error message was just useful while debugging when I unintentionally broke something and caused a difference